lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -1769,13 +1767,16 @@ void GrandCentralPass::runOnOperation() {
   }
 
   // Garbage collect dead NLAs.
+  auto symTable = getSymbolTable();
   for (auto &op :
        llvm::make_early_inc_range(circuitOp.getBody()->getOperations())) {
 
     // Remove NLA operations.
     if (auto nla = dyn_cast<NonLocalAnchor>(op)) {
-      if (deadNLAs.count(nla.sym_nameAttr()))
-        nla.erase();
+      if (deadNLAs.count(nla.sym_nameAttr())) {
+        nlaTable->erase(nla);
```

Comment:
Is this pass preserving the NLATable analysis given that it is doing the erasure inside the NLATable?

Reply:
Thanks for pointing it out, this pass does preserve the NLATable analysis. Updated it.